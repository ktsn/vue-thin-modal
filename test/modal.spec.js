import { mount, createLocalVue } from '@vue/test-utils'
import VueThinModal from '../src'

describe('Modal', () => {
  it('allows to manually mount modal portal when autoMountPortal: false', async () => {
    const localVue = createLocalVue()
    localVue.use(VueThinModal, {
      autoMountPortal: false
    })

    const App = {
      template: `<div>
        <modal name="test">
          <p>{{ message }}</p>
        </modal>
        <modal-portal />
      </div>`,

      data () {
        return {
          message: 'Hello!'
        }
      }
    }

    const wrapper = mount(App, { localVue })
    wrapper.vm.$modal.push('test')
    await wrapper.vm.$nextTick()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
