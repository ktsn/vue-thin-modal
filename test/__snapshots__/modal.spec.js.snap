// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal allows to manually mount modal portal when autoMountPortal: false 1`] = `"<div><!----> <div class=\\"modal-wrapper\\"><div class=\\"modal-backdrop modal-backdrop-enter modal-backdrop-enter-active\\"></div><div role=\\"dialog\\" aria-hidden=\\"false\\" class=\\"modal-content-wrapper\\"><p class=\\"modal-content modal-content-enter modal-content-enter-active\\">Hello!</p></div></div></div>"`;
